engine/class_modules/sc_death_knight.cpp

```diff
@@ -2528,6 +2536,16 @@ struct dancing_rune_weapon_pet_t : public death_knight_pet_t
     int n_targets() const override
     { return p() -> o() -> in_death_and_decay() ? aoe + as<int>( p() -> o() -> spec.death_and_decay_2 -> effectN( 1 ).base_value() ) : aoe; }
 
+    double composite_da_multiplier( const action_state_t* state ) const override
+    {
+    double m = drw_attack_t::composite_da_multiplier( state );
+    if ( p() -> o() -> conduits.withering_plague -> ok() && target ->get_dot( "blood_plague", p()) -> is_ticking() )
```

Comment:
1. I think there is some composite_target_da_multiplier function for this, which I recommend using.
2. This does a string-based dot lookup during combat. Please use death_knight_target_data_t instead. Something like
auto td = get_td(target);
if (td->blood_plague->is_ticking())
  ...

Reply:
You can use composite_target_multiplier, just like in your Debilitating Malady implementation.
Is the damage increase dependant on the player's blood plague being up on the target, or on dancing rune weapon's blood plague? For the former you can fetch the player's target data, for the latter you will have to create a simple target data structure containing the blood plague dot (check how it's done for Magus of the Dead's frostbolt debuff).

Reply:
It is based on DRW's blood plague.  Thanks for the suggestion.  I'll go that route.

Reply:
Ah I did not realize that this was about a pet action, and thus no target data was set up. Looks good now with the added target data class.